Aros changes

deadwood2 opened the pull request:
This patchset contains cumulative changes that happen in AROS repository since ~2013 and were not submitted so far. The changes where compiled for AROS i386, x86_64 and m68k and tested with demo/ programs. No compilation nor testing happened for amigaos3/os4.

If you have any questions or remarks towards patchset let me know!

Commit:
Update muiaros.h from current build.

Commit:
TheButton class AROS changes

Elements of:
64bit fix - pointer types need to be IPTR
Fixes notifcation handling for 64 bit systems
TheBar.mui: 64 bit support and compiler de-lint

Commit:
TheBar class AROS changes

Elements of:
STACKED for x86_64.
64bit fix - pointer types need to be IPTR
TheBar.mui: 64 bit support and compiler de-lint

Commit:
TheBar.mcp class AROS changes

Elements of:
STACKED for x86_64.
TheBar.mui: 64 bit support and compiler de-lint

Commit:
Other AROS changes

jens-maus commented:
I am sorry, but your changes are breaking the other builds (os3, os4, mos, etc.) See here:

https://github.com/amiga-mui/thebar/actions/runs/7856911649/job/21440715456#step:7:17

So please fix.

deadwood2 commented:
Ok, let me rework and resubmit this.

Commit:
Use SIPTR as original code was using LONG. Added define for other platform.

Commit:
Support buttons with MUIA_TheBar_Attr_Sleep == TRUE

Button that is sleeping is still on the list but has obj pointer set to
NULL. Make sure the check does not crash in such situation.

deadwood2 commented:
Hi, is there anything else needed from me to have this PR integrated?
